test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSourcesTests.cs

```diff
@@ -475,6 +475,211 @@ private static void CreateXmlFile(string configFilePath, string nugetConfigStrin
+                    source,
+                    "--name",
+                    "test_source",
+                    "--configfile",
+                    settings.ConfigPath,
+                    "--protocol-version",
+                    "3"
+                };
+
+                // Act
+                var result = _fixture.RunDotnetExpectSuccess(workingPath, string.Join(" ", args));
+
+                // Assert
+                Assert.True(result.ExitCode == 0);
+                var loadedSettings = Settings.LoadDefaultSettings(root: workingPath, configFileName: null, machineWideSettings: null);
+                var packageSourcesSection = loadedSettings.GetSection("packageSources");
+                var sourceItem = packageSourcesSection?.GetFirstItemWithAttribute<SourceItem>("key", "test_source");
+                var expectedProtocolVersion = shouldWriteProtocolVersion ? "3" : null;
+                Assert.Equal(expectedProtocolVersion, sourceItem.ProtocolVersion);
+            }
+        }
+
+        [PlatformTheory(Platform.Windows)]
+        [InlineData("1", false)]
+        [InlineData("2", true)]
+        [InlineData("3", true)]
+        [InlineData("4", false)]
+        [InlineData("5", false)]
+        public void Sources_WhenAddingSourceWithProtocolVersion_ProtocolVersionIsValidated(string protocolVersion, bool shouldSucceed)
+        {
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                var workingPath = pathContext.WorkingDirectory;
+                var settings = pathContext.Settings;
+
+                // Arrange
+                var args = new string[]
+                {
+                    "nuget",
+                    "add",
+                    "source",
+                    "https://source.test",
+                    "--name",
+                    "test_source",
+                    "--configfile",
+                    settings.ConfigPath,
+                    "--protocol-version",
+                    protocolVersion
+                };
+
+                // Act
+                var command = string.Join(" ", args);
+                var result = shouldSucceed ? _fixture.RunDotnetExpectSuccess(workingPath, command) : _fixture.RunDotnetExpectFailure(workingPath, command);
+
+                // Assert
+                if (!shouldSucceed)
+                {
+                    // Assert error message
+                    string expectedErrorMessage = "The protocol version specified is invalid.";
+                    Assert.True(result.Output.Contains(expectedErrorMessage), "Expected error is " + expectedErrorMessage + ". Actual error is " + result.Output);
+                }
+            }
+        }
+
+        [PlatformTheory(Platform.Windows)]
+        [InlineData(@"https://source.test", true)]
+        [InlineData(@"\\myserver\packages", false)]
+        public void Sources_WhenUpdatingSourceWithProtocolVersion_GotUpdated(string source, bool shouldWriteProtocolVersion)
```

Comment:
Like the "add" tests above, using a subdirectory of the `TestDirectory` as a local package source, this can unblock the test from being restricted to Windows only.